xdsl/interpreters/builtin.py

```diff
@@ -53,23 +54,41 @@ def run_cast(
             for (o, r, arg) in zip(op.operands, op.results, args)
         )
 
-    @impl_attr(FloatAttr)
-    def float_attr_value(self, interpreter: Interpreter, attr: AnyFloatAttr) -> float:
+    @impl_attr(Float64Type)
+    def float64_attr_value(
+        self, interpreter: Interpreter, attr: Attribute, attr_type: Float64Type
+    ) -> float:
+        interpreter.interpreter_assert(isa(attr, AnyFloatAttr))
+        attr = cast(AnyFloatAttr, attr)
+        return attr.value.data
+
+    @impl_attr(Float32Type)
+    def float32_attr_value(
+        self, interpreter: Interpreter, attr: Attribute, attr_type: Float32Type
+    ) -> float:
+        interpreter.interpreter_assert(isa(attr, AnyFloatAttr))
+        attr = cast(AnyFloatAttr, attr)
         return attr.value.data
 
-    @impl_attr(IntegerAttr)
+    @impl_attr(IntegerType)
     def integer_attr_value(
```

Comment:
Should this be `integer_type_value`/`integer_type_attr_value` then?

Reply:
At the end of the day, it's kind of equivalent, since it can only handle integer attrs